consensus/beacon/consensus.go

```diff
@@ -130,6 +132,14 @@ func (beacon *Beacon) VerifyHeaders(chain consensus.ChainHeaderReader, headers [
 			oldDone, oldResult = beacon.ethone.VerifyHeaders(chain, preHeaders, preSeals)
 			newDone, newResult = beacon.verifyHeaders(chain, postHeaders, preHeaders[len(preHeaders)-1])
 		)
+		// verify that the headers are valid wrt. the terminal block.
+		index, err := beacon.verifyTerminalPoWBlock(chain, preHeaders)
+		if err != nil {
+			// Mark all subsequent headers with the error.
+			for i := index; i < len(preHeaders); i++ {
+				errors[i], done[i] = err, true
```

Comment:
The fault is that you set the result for `errors/done[len(preHeaders)-1]`, then below you skip that because it's already set, causing the last verifier from preHeaders to overflow into postHeader results.

Please don't hack this, do it the right way. Split the thing into preHeaders, transitionHeader and postHeaders. Then you have 3 verification for the 3 different things.

Reply:
Hmm, maybe not, in that case, you overwrite the ttd error with the rthone verified when that arrives from `oldResults`. Either way, it's messy.